App_Data/SwaggerSpecs/Microsoft.GuestConfiguration/guestconfiguration.json

```diff
@@ -0,0 +1,1285 @@
+          "description": "List all guest configuration assignments for a virtual machine.",
+          "x-ms-examples": {
+            "List all guest configuration assignments for a virtual machine": {
+              "$ref": "./examples/listGuestConfigurationAssignments.json"
+            }
+          },
+          "parameters": [
+            {
+              "$ref": "../../common/v1/definitions.json#/parameters/ResourceGroupNameParameter"
+            },
+            {
+              "$ref": "../../common/v1/definitions.json#/parameters/SubscriptionIdParameter"
+            },
+            {
+              "$ref": "../../common/v1/definitions.json#/parameters/VmNameParameter"
+            },
+            {
+              "$ref": "../../common/v1/definitions.json#/parameters/ApiVersionParameter"
+            }
+          ],
+          "responses": {
+            "200": {
+              "description": "OK",
+              "schema": {
+                "$ref": "#/definitions/GuestConfigurationAssignmentList"
+              }
+            },
+            "default": {
+              "description": "Error response describing why the operation failed.",
+              "schema": {
+                "$ref": "../../common/v1/definitions.json#/definitions/ErrorResponse"
+              }
+            }
+          },
+          "x-ms-pageable": {
+            "nextLinkName": null
+          },
+          "produces": [
+            "application/json"
+          ],
+          "consumes": [
+            "application/json"
+          ]
+        }
+      },
+      "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Compute/virtualMachines/{vmName}/providers/Microsoft.GuestConfiguration/guestConfigurationAssignments/{guestConfigurationAssignmentName}/reports": {
+        "get": {
+          "tags": [
+            "GuestConfigurationAssignmentReports"
+          ],
+          "operationId": "GuestConfigurationAssignmentReports_List",
+          "description": "List all reports for the guest configuration assignment, latest report first.",
+          "x-ms-examples": {
+            "List all guest configuration assignments for a virtual machine": {
+              "$ref": "./examples/listAllGuestConfigurationAssignmentReports.json"
+            }
+          },
+          "parameters": [
+            {
+              "$ref": "../../common/v1/definitions.json#/parameters/ResourceGroupNameParameter"
+            },
+            {
+              "name": "guestConfigurationAssignmentName",
+              "in": "path",
+              "required": true,
+              "type": "string",
+              "description": "The guest configuration assignment name."
+            },
+            {
+              "$ref": "../../common/v1/definitions.json#/parameters/SubscriptionIdParameter"
+            },
+            {
+              "$ref": "../../common/v1/definitions.json#/parameters/VmNameParameter"
```

Comment:
Looks like some of these types are not part of the global common definitions file
https://github.com/Azure/azure-rest-api-specs/tree/d77baea351ade4453199f73e9aad1f4b8a320a76/specification/common-types

and instead of specific to Microsoft.GuestConfiguration at
https://github.com/Azure/azure-rest-api-specs/blob/d77baea351ade4453199f73e9aad1f4b8a320a76/specification/guestconfiguration/resource-manager/Microsoft.GuestConfiguration/common/v1/definitions.json

In that case we would need to copy over that definitions file as well (and update the path to the definitions file)

Reply:
got it. do I need a folder structure?